verilog/tools/syntax/export_json_examples/verible_verilog_syntax_test.py

```diff
@@ -0,0 +1,271 @@
+    self.assertIsNotNone(data)
+    for f in files:
+      self.assertIsNotNone(data[f.name])
+
+    self.assertIsNotNone(data[files[0].name].errors)
+    self.assertIsNone(data[files[1].name].errors)
+
+
+class TestTree(VeribleVerilogSyntaxTest):
+  def setUp(self):
+    super().setUp()
+    data = self.parser.parse_string(
+        "module ModuleName#(parameter PARAM_NAME=42)\n" +
+        "    (input portIn, output portOut);\n" +
+        "  import import_pkg_name::*;\n" +
+        "  wire wireName;\n" +
+        "endmodule;\n" +
+        "module OtherModule; endmodule;\n")
+    self.assertIsNotNone(data)
+    self.assertIsNotNone(data.tree)
+    self.tree = data.tree
+
+  def test_find(self):
+    header = self.tree.find({"tag": "kModuleHeader"})
+    self.assertIsNotNone(header)
+    module_name = header.find({"tag": "SymbolIdentifier"})
+    self.assertIsNotNone(module_name)
+    self.assertEqual(module_name.text, "ModuleName")
+    nonexistent = header.find({"tag": "SomeUndefinedTag"})
+    self.assertIsNone(nonexistent)
+
+  def test_find_all(self):
+    headers = self.tree.find_all({"tag": "kModuleHeader"})
+    self.assertEqual(len(headers), 2)
+
+    identifiers = self.tree.find_all({"tag": "SymbolIdentifier"})
+    self.assertEqual(len(identifiers), 7)
+
+    some_identifiers = self.tree.find_all({"tag": "SymbolIdentifier"},
+                                          max_count=4)
+    self.assertEqual(len(some_identifiers), 4)
+
+  def test_iter_find_all(self):
+    identifiers = [n.text
+                   for n
+                   in self.tree.iter_find_all({"tag": "SymbolIdentifier"})]
+    self.assertIn("ModuleName", identifiers)
+    self.assertIn("PARAM_NAME", identifiers)
+    self.assertIn("OtherModule", identifiers)
+
+  def test_custom_filter(self):
+    def tokens_past_135_byte(node):
+      return (isinstance(node, verible_verilog_syntax.TokenNode)
+              and node.start > 135)
+
+    other_module_tokens = self.tree.find_all(tokens_past_135_byte)
+    self.assertGreaterEqual(len(other_module_tokens), 5)
+    for token in other_module_tokens:
+      self.assertGreater(token.start, 135)
+
+  def test_search_order(self):
+    level_order = self.tree.find_all({"tag": "SymbolIdentifier"})
+    depth_order = self.tree.find_all({"tag": "SymbolIdentifier"},
+                              iter_=verible_verilog_syntax.PreOrderTreeIterator)
+
+    def check_items_order(iterable, items_to_check):
+      index = 0
+      for item in iterable:
+        if items_to_check[index] == item:
+          index += 1
+        if index == len(items_to_check):
+          return True
+      return False
+
+    self.assertTrue(check_items_order([n.text for n in level_order],
+                                      ["ModuleName", "OtherModule", "portIn"]))
+    self.assertTrue(check_items_order([n.text for n in depth_order],
+                                      ["ModuleName", "portIn", "OtherModule"]))
```

Comment:
Will `assertSequenceEqual` work here?
https://docs.python.org/3/library/unittest.html#unittest.TestCase.assertSequenceEqual

Reply:
No. `assertSequenceEqual` tests for exact sequence, while the code needs to ignore extra tokens between tested elements.
I've decided to not test the whole sequence, because the order of some tokens depends on CST implementation details (e.g. branches depth).